galsim/wfirst/wfirst_psfs.py

```diff
@@ -0,0 +1,460 @@
+                                   users who want to interpolate can always set up the interpolation
+                                   later on even if they do not do so when calling getPSF().
+                                   [default: None]
+    @param    extra_aberrations    Array of extra aberrations to include in the PSF model, on top of
+                                   those that are part of the WFIRST design.  These should be
+                                   provided in units of waves at the fiducial wavelength of 1293 nm,
+                                   as an array of length 12 with entries 4 through 11 corresponding
+                                   to defocus through spherical aberrations.  [default: None]
+    @param    wavelength_limits    A tuple or list of the blue and red wavelength limits to use for
+                                   interpolating the chromatic object, if `n_waves` is not None.  If
+                                   None, then it uses the blue and red limits of all imaging
+                                   passbands to determine the most inclusive wavelength range
+                                   possible.  But this keyword can be used to reduce the range of
+                                   wavelengths if only one passband (or a subset of passbands) is to
+                                   be used for making the images.
+                                   [default: None]
+    @param    verbose              Should the routine provide information about its progress in
+                                   loading the per-SCA PSFs?  This can be valuable in tracking
+                                   progress, since the full calculations including the real WFIRST
+                                   pupil plane can be quite expensive.  [default: True]
+    @param    achromatic           An option to get an achromatic PSF for a single wavelength, for
+                                   users who do not care about chromaticity of the PSF.  If False,
+                                   then the fully chromatic PSF is returned.  Alternatively the user
+                                   should supply either (a) a wavelength in nanometers, and they
+                                   will get a list of achromatic OpticalPSF objects for that
+                                   wavelength, or (b) a bandpass object, in which case they will get
+                                   a list of achromatic OpticalPSF objects defined at the effective
+                                   wavelength of that bandpass.
+                                   [default: False]
+    @returns  A list of ChromaticOpticalPSF objects, one for each SCA.  The SCAs in WFIRST are
+              1-indexed, and the list is indexed according go the SCA, meaning that the 0th object
+              in the list is always None, the 1st is the PSF for SCA 1, and so on.
```

Comment:
This is confusing.  If I ask for `getPSF(SCAs = [3,4,5])`, I would expect to get back a list of 3 PSF objects.  Not a list of 19 items, 16 of which are `None`.

Perhaps a better return type would be a dict, where the keys are the same as the input `SCAa`.  You would still use it the same way, but the returned object would be more obvious in terms of its meaning.